Classes/Private/Manager/Video/OCTVideoEngine.m

```diff
@@ -0,0 +1,246 @@
+//
+//  OCTVideoEngine.m
+//  objcTox
+//
+//  Created by Chuong Vu on 6/19/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "OCTVideoEngine.h"
+#import "OCTVideoView.h"
+#import "DDLog.h"
+
+#undef LOG_LEVEL_DEF
+#define LOG_LEVEL_DEF LOG_LEVEL_VERBOSE
+
+@import AVFoundation;
+
+static uint8_t *reusableUChromaPlane;
+static uint8_t *reusableVChromaPlane;
+
+static const OSType kPixelFormat = kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange;
+
+@interface OCTVideoEngine () <AVCaptureVideoDataOutputSampleBufferDelegate>
+
+@property (nonatomic, strong) AVCaptureSession *captureSession;
+@property (nonatomic, strong) AVCaptureVideoDataOutput *dataOutput;
+@property (nonatomic, strong) dispatch_queue_t processingQueue;
+@property (nonatomic, strong) OCTVideoView *videoView;
+
+@property (nonatomic, assign) NSUInteger sizeOfChromaPlanes;
+
+@end
+
+@implementation OCTVideoEngine
+
+- (instancetype)init
+{
+    DDLogVerbose(@"%@: init", self);
+    self = [super init];
+    if (! self) {
+        return nil;
+    }
+
+    _captureSession = [AVCaptureSession new];
+    _captureSession.sessionPreset = AVCaptureSessionPreset1280x720;
```

Comment:
For the future - this should be. It would be nice to change quality depending of bandwidth.